analyzers/src/SonarAnalyzer.CSharp/Rules/EqualityOnFloatingPoint.cs

```diff
@@ -49,85 +51,75 @@ protected override void Initialize(SonarAnalysisContext context)
         private static void CheckLogicalExpression(SonarSyntaxNodeReportingContext context)
         {
             var binaryExpression = (BinaryExpressionSyntax)context.Node;
-            var left = TryGetBinaryExpression(binaryExpression.Left);
-            var right = TryGetBinaryExpression(binaryExpression.Right);
-
-            if (right == null || left == null)
-            {
-                return;
-            }
-
-            var eqRight = CSharpEquivalenceChecker.AreEquivalent(right.Right, left.Right);
-            var eqLeft = CSharpEquivalenceChecker.AreEquivalent(right.Left, left.Left);
-            if (!eqRight || !eqLeft)
-            {
-                return;
-            }
-
-            var isEquality = IsIndirectEquality(context.SemanticModel, binaryExpression, left, right);
 
-            if (isEquality || IsIndirectInequality(context.SemanticModel, binaryExpression, left, right))
+            if (TryGetBinaryExpression(binaryExpression.Left) is { } left
+                && TryGetBinaryExpression(binaryExpression.Right) is { } right
+                && (CSharpEquivalenceChecker.AreEquivalent(right.Right, left.Right) && CSharpEquivalenceChecker.AreEquivalent(right.Left, left.Left))
```

Comment:
Having `&&` needs this formatting
```suggestion
                && CSharpEquivalenceChecker.AreEquivalent(right.Right, left.Right) 
                && CSharpEquivalenceChecker.AreEquivalent(right.Left, left.Left)
```